fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSessionTest.java

```diff
@@ -195,4 +223,243 @@ public void getTriplesOfBinaryDescription() throws Exception {
+
+        final Node resourceUri = createURI(resourceId);
+
+        when(rdfSourceOperation.getTriples()).thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getResourceId()).thenReturn(resourceId);
+        when(rdfSourceOperation.getType()).thenReturn(CREATE);
+
+        final Node resourceUri2 = createURI(resourceId2);
+
+        when(rdfSourceOperation2.getTriples()).thenReturn(new DefaultRdfStream(resourceUri2, Stream.empty()));
+        when(rdfSourceOperation2.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri2, Stream.empty()));
+        when(rdfSourceOperation2.getResourceId()).thenReturn(resourceId2);
+        when(rdfSourceOperation2.getType()).thenReturn(CREATE);
+
+        //mock success on commit for the first object session
+        when(mockSessionFactory.create(eq(OCFL_OBJECT_ID), anyString())).thenReturn(objectSession1);
+        when(objectSession1.commit(eq(CommitOption.UNVERSIONED))).thenReturn(Instant.now().toString());
+        //mock failure on commit for the second object session
+        when(mockSessionFactory.create(eq(ocfObjectId2), anyString())).thenReturn(objectSession2);
+        when(objectSession2.commit(eq(CommitOption.UNVERSIONED))).thenThrow(PersistentStorageException.class);
+
+        final PersistentStorageSession session1 = createSession(index, mockSessionFactory);
+        //persist the two operations
+        session1.persist(rdfSourceOperation);
+
+        //perform the create rdf operation
+        session1.persist(rdfSourceOperation2);
+
+        //get triples should now fail because the session is effectively closed.
+        try {
+            session1.commit(CommitOption.UNVERSIONED);
+            fail("session1.commit(...) invocation should fail.");
+        } catch (PersistentStorageException ex) {
+            //attempted rollback should also fail:
+            try {
+                session1.rollback();
+                fail("session1.rollback(...) invocation should fail.");
+            } catch (final PersistentStorageException e) {
+                assertTrue("exception does not contain expected text",
+                        e.getMessage().contains("already committed to the unversioned "));
+            }
+        }
+    }
+
+    @Test
+    public void getTriplesFailsIfCommitHasAlreadyStarted() throws PersistentStorageException {
+        when(mapping.getOcflObjectId()).thenReturn(OCFL_OBJECT_ID);
+        when(mapping.getParentFedoraResourceId()).thenReturn(parentId);
+        when(index.getMapping(resourceId)).thenReturn(mapping);
+
+        final Node resourceUri = createURI(resourceId);
+
+        when(rdfSourceOperation.getTriples()).thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
```

Comment:
These four statements are in most of the tests, sometimes more than once. Maybe you can break it out into a well named helper method that takes a rdfSourceOperation to help with readability/length?

Reply:
+1